src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -1363,16 +1363,25 @@ public ReceivedMessage createMessage(DataObject jsonObject, @Nullable MessageCha
         final long authorId = author.getLong("id");
         MemberImpl member = null;
 
-        if (channel == null && jsonObject.isNull("guild_id"))
+        if (jsonObject.isNull("guild_id"))
         {
-            DataObject channelData = DataObject.empty()
-                    .put("id", channelId);
-            //if we see an author that isn't us, we can assume that is the other side of this private channel
-            //if the author is us, we learn no information about the user at the other end
-            if (authorId != getJDA().getSelfUser().getIdLong())
+            //we know it's a private channel
+            PrivateChannelImpl priv = (PrivateChannelImpl) channel;
```

Comment:
Wouldn't this cause an exception when `channel` is null?

Reply:
No.

Reply:
Are you sure?
To my knowledge does casting a null object cause an exception. And unless I miss something is `channel` nullable, no?

Reply:
```
jshell> (String) null
$1 ==> null
```